content/blog/monkey-patch.md

```diff
@@ -0,0 +1,62 @@
++++
+title = "Monkey Patching"
+slug = "monkey-patch"
+date = 2023-10-17T18:29:54+05:30
+image = "/images/2023/10/monkeypatch.jpg"
+draft = false
+authors = ["Nitin Rajkumar"]
+description = "Monkey patching and it's sideeffects in Ruby"
+tags = ["Software Craftsmanship", "GraphQL"]
+categories = ["Software Craftsmanship", "Ruby", "Monkey Patching"]
+type = ""
++++
+
+Ruby, being a dynamic programming language provides the ability and hence freedom to reopen the existing classes and change their behavior during runtime by adding new methods or modifying existing methods. 
+
+#### How to monkey patch
+
+```ruby
+class Sound
+	def car
+		"honk"
+	end
+end
+class Sound
+	def bike
+		"peep-peep"
+	end
+end
+noise = Sound.new
+noise.car #=> honk
+noise.bike #=> peep-peep
```

Comment:
Either rename the class to Noise or rename the variable to sound

Noise is a special category of Sound
Not all sounds are noise

so Sound.new cannot be called a noise
```suggestion
sound = Sound.new
sound.car #=> honk
sound.bike #=> peep-peep
```